### PR TITLE
Add Project Listing && Update Project

### PR DESCRIPTION
**Pull request by queq1890**
## Story URL
https://trello.com/c/KwHTjffL/48-2-sprint5project-listing
https://trello.com/c/w94oMhPe/49-3-sprint5edit-project

## What I did 
- Add Project Listing Page
- Add Project Update Form

### CODE REVIEW

File: src/components/ProjectDelete/index.js [hunk 1]
```diff
@@ -0,0 +1,70 @@
+import React, { Fragment, Component } from 'react';
+import { Button, Icon, Modal } from 'semantic-ui-react';
+import { IconBtn } from '../../styles/ContentsTable';
+
+// This component inmplementation is not finished yet
+// TODO : Add Delete Mutation to handle delete project
```
Comment (queq1890) [inline src/components/ProjectDelete/index.js:6]: 🚨
I created this component but the Project Delete feature is not implemented yet.
I will implement in next pull request.

File: src/containers/TeamPage/DailyReportContainer.js [hunk 2]
```diff
@@ -60,22 +60,23 @@ class DailyReportContainer extends Component {
```
Comment (queq1890) [inline src/containers/TeamPage/DailyReportContainer.js:60]: (Not related to This pull request directly )

Fixed this component based on the new daily report structure
- delete achievement
- Add tasks field

File: src/components/ProjectDelete/index.js [hunk 3]
```diff
@@ -0,0 +1,70 @@
...
+              content="Yes"
+            />
+          </Modal.Actions>
+        </Modal>
+      </Fragment>
+    );
+  }
+}
+
+export default ProjectDelete;
```
Comment (phuongthuan) [inline src/components/ProjectDelete/index.js:70]: Right now, we have 2 modal confirm. In the future, If we use Modal Confirm more than two times, let consider create a `ModalConfirm` component

File: src/components/ProjectFormUpdate/index.js [hunk 4]
```diff
@@ -0,0 +1,190 @@
...
+  return {
+    label: name,
+    value: id
+  };
+};
+
+const ProjectFormUpdate = ({ match }) => (
+  <Query query={SINGLE_PROJECT_QUERY} variables={{ id: match.params.id }}>
+    {({ data: projectData, error: projectError, loading: projectLoading }) => {
+      if (projectLoading) return <Loader size="tiny" active inline="centered" />;
```
Comment (phuongthuan) [inline src/components/ProjectFormUpdate/index.js]: Wh not use `Spinner` component?

File: src/components/ProjectListing/index.js [hunk 5]
```diff
@@ -0,0 +1,86 @@
...
+import { ProjectListHeader, ProjectListRow } from '../../styles/ProjectList';
+import history from '../../utils/history';
+
+const headerItems = ['No', 'title', 'team leader', 'members', 'actions'];
+
+const ProjectListing = () => (
+  <ContentWrapper>
+    <Query query={ALL_PROJECTS_QUERY}>
+      {({ data, loading, error }) => {
+        if (loading) return <Loader size="tiny" active inline="centered" />;
```
Comment (phuongthuan) [inline src/components/ProjectListing/index.js]: Same above

File: src/components/ProjectListing/index.js [hunk 6]
```diff
@@ -0,0 +1,86 @@
...
+                <ContentsRowColumn onClick={() => history.push(`/projects/${project.id}`)}>
+                  <a>{project.title}</a>
+                </ContentsRowColumn>
+                <ContentsRowColumn>
+                  <Image src={project.teamLeader.avatar} size="mini" avatar />
+                  {project.teamLeader.name}
+                </ContentsRowColumn>
+                <ContentsRowColumn>
+                  {project.members.map(member => (
+                    <span key={uuidv1()}>{member.name} </span>
```
Comment (phuongthuan) [inline src/components/ProjectListing/index.js]: shouldn't abuse `uuid` while `id` of member can be a key.

**Review by phuongthuan — APPROVED** [review]
A have some comment

File: src/components/ProjectFormUpdate/index.js (hunk 4 above)
Comment (queq1890) [reply src/components/ProjectFormUpdate/index.js]: @phuongthuan 
Could you give me a doc or standard for which spinner or loader should be used in some situations? Without that , I can't figure out which component is proper

File: src/components/ProjectDelete/index.js (hunk 3 above)
Comment (queq1890) [reply src/components/ProjectDelete/index.js:70]: That makes sense, I'll add `components/ModalConfirm` or something 👍

File: src/components/ProjectFormUpdate/index.js (hunk 4 above)
Comment (phuongthuan) [reply src/components/ProjectFormUpdate/index.js]: The `Spinner` component just basically a plain HTML css component. 
All you need to do is just use it, nothing special:
`if (loading) return <Spinner/>;`

Comment (queq1890) [reply src/components/ProjectFormUpdate/index.js]: I know that, I'm asking when we use 'em ?
- For Page Content loading ... `<Spinner />` ?
- For other tiny components loading ... `<Loader />` ? 

I don't like to be criticized when the standard of a proper situation for each component is unclear.
I'll add comment out on `<Spinner />` component.

**Pull request merged**